dingo/core/samplers.py

```diff
@@ -492,10 +492,14 @@ def _run_sampler(
 
             time_sample_start = time.time()
             self.model.model.eval()
+
             with torch.no_grad():
-                y, log_prob = self.model.model.sample_and_log_prob(
-                    x["data"], x["context_parameters"]
-                )
+                if "context_parameters" not in x.keys():
+                    y, log_prob = self.model.model.sample_and_log_prob(x["data"])
+                else:
+                    y, log_prob = self.model.model.sample_and_log_prob(
+                        x["data"], x["context_parameters"]
+                    )
```

Comment:
I would swap the order of the `if`/`else` so you have fewer negatives, and also you don't need the `.keys()`:
```suggestion
                if "context_parameters" in x:
                    y, log_prob = self.model.model.sample_and_log_prob(
                        x["data"], x["context_parameters"]
                    )
                else:
                    y, log_prob = self.model.model.sample_and_log_prob(x["data"])
```